packages/core/src/odata/common/uri-conversion/get-filter.ts

```diff
@@ -130,6 +134,42 @@ export function createGetFilter(uriConverter: UriConverter) {
         uriConverter.convertToUriFormat(filter.value, filter.edmType!)
       ].join(' ');
     }
+
+    if (isFilterLambdaExpression(filter)) {
+      return getODataFilterExpressionWhenBeingFilterLambdaExpression(
+        filter,
+        parentFieldNames,
+        targetEntityConstructor
+      );
+    }
+  }
+
+  function getODataFilterExpressionWhenBeingFilterLambdaExpression<
+    FilterEntityT extends EntityBase
+  >(
+    filter: FilterLambdaExpression<FieldType>,
+    parentFieldNames: string[] = [],
+    targetEntityConstructor: Constructable<any>
+  ) {
+    const alias = 'a';
+    if (typeof filter.innerFilter.field !== 'string') {
+      throw new Error(
+        `The type of the field: ${
+          filter.innerFilter.field
+        } is not string, but ${typeof filter.innerFilter.field}.`
+      );
+    }
+    const field = retrieveField(
+      filter.innerFilter.field,
+      targetEntityConstructor,
+      filter.innerFilter.edmType
+    );
+    const filterExp = [
+      [...parentFieldNames, filter.innerFilter.field].join('/'),
+      filter.innerFilter.operator,
+      uriConverter.convertToUriFormat(filter.innerFilter.value, field.edmType)
+    ].join(' ');
+    return `${filter.rootFiledName}/${filter.lambdaOperator}(${alias}:${alias}/${filterExp})`;
```

Comment:
I do not see a test regarding the URI creation. IN the filter-factory.ts I se the string `"(to_MultiLink/any(a:a/StringProperty eq 'test1') and to_MultiLink/all(a:a/StringProperty eq 'test2'))"` hard coded but would be good to have a test for this functionality.